pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,6 +364,39 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
         obj.value += nanos
         obj.dts.ps = nanos * 1000
 
+    obj.fold = 0
+    if tz is not None:
```

Comment:
Can this logic be moved to a function? The flow looks similar to https://github.com/pandas-dev/pandas/pull/31563/files#diff-66d289312fd0b02e7721bf45fb281797R437 and https://github.com/pandas-dev/pandas/pull/31563/files#diff-66d289312fd0b02e7721bf45fb281797R633

Reply:
Moved and pinged you an an example.